rankboost.py

```diff
@@ -436,6 +436,42 @@ def prepare_dataframe(mhc, predfile, expfile, pepfile, efilt_threshold, boost):
     _log.info('Finished creating the data frame for rank boosting..')
     return all_data, stats, boost
 
+def collapse_iars (stats):
+    """
+    Collapse duplicated IARs.
+    :param stats:
+    :return:
+    """
+    for pep_name_o in stats.index:
```

Comment:
This whole approach is very dicey... you are modifying a list while you are iterating through it. Hence the numerous `try: except KeyError:`s. Instead, if you group by gene name (related IARS are GUARANTEED to be from the same gene), if len(group) = 1 then add to output dataframe, else collapse.